src/bin/hexyl.rs

```diff
@@ -161,10 +178,127 @@ fn main() {
     }
 }
 
-fn parse_hex_or_int(n: &str) -> Option<u64> {
-    if n.starts_with("0x") {
-        u64::from_str_radix(n.trim_start_matches("0x"), 16).ok()
-    } else {
-        n.parse::<u64>().ok()
+fn parse_byte_count(n: &str, block_size: u64) -> Option<u64> {
+    const HEX_PREFIX: &'static str = "0x";
+
+    let n = {
+        let mut chars = n.chars();
+        match chars.next()? {
+            '+' => chars.as_str(),
+            _ => n,
+        }
+    };
+
+    if n.starts_with(HEX_PREFIX) {
+        let n = &n[HEX_PREFIX.len()..];
+        if !n.chars().all(|c| match c {
```

Comment:
this validation can be skipped, `u64::from_str_radix` will do it anyway

Reply:
From the commit message:

> This is different from simply using `from_str_radix` because `from_str_radix` accepts signs (in this case, only `'+'`).

Reply:
Thinking about this some more, it would probably be just as well to check for a `'+'` as the first character, and let `u64::from_str_radix` fail the other cases. That would better match the intent of this code.

Reply:
Changed. What do you think now? :)